Fixes labels/bad input/example input for Improve usability of the DriverAvailabilities Create Page #13

Tommygithubaccount123 opened the pull request:
[DOKKU](https://tommy-dev.dokku-13.cs.ucsb.edu/)

Does part of EPIC issue [13](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-5/issues/13). 

Closes #32  (Made day of week a dropdown, only able to select day of week)

Note: Did not change driverId since we will remove this later on in the EPIC. Only front-end changes.

- The form labels are improved
- The bad input and missing input error messages are added and improved
- Example inputs on each form field
- Dropdown for choosing day instead of free form

BEFORE:
![image](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-5/assets/79723996/d7051c08-4f5d-4cc4-b43e-f0e5cbb03c83)

![image](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-5/assets/79723996/9f94b7ec-37f0-4e78-9b41-f1e30722608d)


AFTER

![image](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-5/assets/79723996/a05756b2-7001-4432-bc3d-dfca1278181b)
![image](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-5/assets/79723996/369dd3bf-544a-450d-9818-4753d4fd9d96)
![image](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-5/assets/79723996/f7ff7f69-257a-4518-b33f-19452a20a758)


Code modeled from `RideForm.js`

Tommygithubaccount123 committed:
changes to labels

Tommygithubaccount123 committed:
fix tests

Tommygithubaccount123 committed:
fix page tests

Tommygithubaccount123 committed:
fix stryker comment

Tommygithubaccount123 committed:
add notes example

Ononymous reviewed (dismissed):
LGTM

pconrad commented on frontend/src/main/components/Driver/DriverAvailabilityForm.js:
```diff
@@ -11,7 +11,7 @@ function DriverAvailabilityForm({ initialContents, submitAction, buttonLabel = "
         { defaultValues: initialContents || {}, }
     );
-    // Stryker restore all
+    // Stryker enable all
```
```suggestion
    // Stryker restore all
```

It turns out that `    // Stryker enable all` isn't a thing.  We need to change all occurences of `// Stryker enable all` to `// Stryker restore all` as find them.

pconrad reviewed (changes requested):
One small change needed.

Also, please deploy this on Dokku and link to the dokku deployment so we can test it more thoroughly.

Tommygithubaccount123 committed:
Update frontend/src/main/components/Driver/DriverAvailabilityForm.js

Co-authored-by: Phill Conrad <pconrad@cs.ucsb.edu>

Tommygithubaccount123 committed:
Added testing

Tommygithubaccount123 commented:
I added tests since the changing the stryker comment made the mutations happen. I added [dokku](https://tommy-dev.dokku-13.cs.ucsb.edu/) link in PR

@pconrad

andrewpengucsb commented on frontend/src/main/components/Driver/DriverAvailabilityForm.js:
```diff
@@ -47,63 +47,85 @@ function DriverAvailabilityForm({ initialContents, submitAction, buttonLabel = "
                     isInvalid={Boolean(errors.notes)}
                     {...register("notes", {
-                        required: "notes is required."
+                        required: "Notes are required."
```
I must have not realized this when writing the issue, but small change...

I don't think that the notes field should be required since, there may be no need for a driver to add any notes.

andrewpengucsb replied:
Dont leave commented code

Tommygithubaccount123 committed:
TS - remove notes requirement

Tommygithubaccount123 committed:
remove comments

walexgreen reviewed (approved):
LGTM

andrewpengucsb reviewed (approved):
LGTM
